server/liveblog/themes/themes.py

```diff
@@ -77,43 +96,126 @@ class ThemesResource(Resource):
 class ThemesService(BaseService):
 
     def get_local_themes_packages(self):
-            embed_folder = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                        '../', 'embed', 'embed_assets', 'themes')
-            for file in glob.glob(embed_folder + '/**/theme.json'):
-                yield json.loads(open(file).read())
-
-    def update_registered_theme_with_local_files(self):
-        created = []
-        updated = []
-        for theme in self.get_local_themes_packages():
-            previous_theme = self.find_one(req=None, name=theme.get('name'))
-            if previous_theme:
+        theme_folder = os.path.join(CURRENT_DIRECTORY, ASSETS_DIR)
+        for file in glob.glob(theme_folder + '/**/theme.json'):
+            files = []
+            for root, dirnames, filenames in os.walk(os.path.dirname(file)):
+                for filename in filenames:
+                    files.append(os.path.join(root, filename))
+            yield json.loads(open(file).read()), files
+
+    def update_registered_theme_with_local_files(self, force=False):
+        results = {'created': [], 'updated': []}
+        for theme, files in self.get_local_themes_packages():
+            result = self.save_or_update_theme(theme, files, force_update=force)
+            results[result.get('status')].append(result.get('theme'))
+        return (results.get('created'), results.get('updated'))
+
+    def save_or_update_theme(self, theme, files=[], force_update=False):
+        # Save the file in the media storage if needed
+        for name in files:
+            with open(name, 'rb') as file:
+                if name.endswith('screenshot.png') or type(app.media).__name__ is 'AmazonMediaStorage':
+                    # set the content type
+                    content_type = mime.from_file(name).decode('utf8')
+                    if content_type == 'text/plain' and name.endswith(tuple(CONTENT_TYPES.keys())):
+                        content_type = CONTENT_TYPES[os.path.splitext(name)[1]]
+                    final_file_name = os.path.relpath(name, CURRENT_DIRECTORY)
+                    # remove existing first
+                    app.media.delete(final_file_name)
+                    # upload
+                    file_id = app.media.put(file.read(), filename=final_file_name, content_type=content_type)
+                    # save the screenshot url
+                    if name.endswith('screenshot.png'):
```

Comment:
Yes, as wordpress does, we expect that themes contain a preview in a file named `screenshot.png`.

The theme structure and the screenshot dimensions will be defined into the documentation that explains how to create a theme.